File: src/main/java/wraith/fabricaeexnihilo/modules/sieves/MeshItem.java
package wraith.fabricaeexnihilo.modules.sieves;

import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;
import net.minecraft.text.Text;
import wraith.fabricaeexnihilo.FabricaeExNihilo;
import wraith.fabricaeexnihilo.modules.base.Colored;
import wraith.fabricaeexnihilo.util.Color;

public class MeshItem extends Item implements Colored {
    private final Color color;
    private final int enchantability;
    
    public MeshItem(Color color, int enchantability, Settings settings) {
        super(settings.maxCount(FabricaeExNihilo.CONFIG.modules.sieves.meshStackSize));
        this.color = color;
        this.enchantability = enchantability;
    }
    
    @Override
    public int getColor(int index) {
        return color.toInt();
    }
    
    @Override
    public boolean isEnchantable(ItemStack stack) {
        return true;
    }
    
    @Override
    public int getEnchantability() {
        return this.enchantability;
    }
    
    public Text getName(ItemStack stack) {
        return getName();
    }
    
}

File: src/main/kotlin/exnihilofabrico/modules/sieves/MeshItem.kt
package exnihilofabrico.modules.sieves

import exnihilofabrico.ExNihiloFabrico
import exnihilofabrico.modules.base.BaseItem
import exnihilofabrico.modules.base.IHasColor
import exnihilofabrico.util.Color
import net.fabricmc.api.EnvType
import net.fabricmc.api.Environment
import net.minecraft.item.Item
import net.minecraft.item.ItemStack
import net.minecraft.text.Text
import net.minecraft.text.TranslatableText

class MeshItem(val color: Color, val displayName: String, settings: Settings = itemSettings): BaseItem(settings), IHasColor {

    override fun getColor(index: Int) = color.toInt()

    @Environment(EnvType.CLIENT)
    override fun getName(): Text {
        return TranslatableText("item.exnihilofabrico.mesh", TranslatableText(displayName))
    }
    override fun getName(stack: ItemStack): Text {
        return TranslatableText("item.exnihilofabrico.mesh", TranslatableText(displayName))
    }

    companion object {
        val itemSettings = Item.Settings().group(ExNihiloFabrico.ITEM_GROUP).maxCount(ExNihiloFabrico.config.modules.sieves.meshStackSize)
    }
}